hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -174,15 +174,18 @@
             readCommittedScope);
 
     // process and resolve the expression
-    Expression filter = getFilterPredicates(job.getConfiguration());
+    DataMapFilter dataMapFilter = getFilterPredicates(job.getConfiguration());
     // this will be null in case of corrupt schema file.
     PartitionInfo partitionInfo = carbonTable.getPartitionInfo(carbonTable.getTableName());
 
+    if (dataMapFilter != null) {
+      dataMapFilter.resolve(false);
```

Comment:
In data map filter class add a method which will resolve and process

Reply:
done